glusterd2/gdctx/global.go

```diff
@@ -94,18 +94,35 @@ func GenerateLocalAuthToken() error {
 }
 
 func protectAuthFile(authfile string) error {
+	var (
+		uGID       string
+		unknownGrp bool
+	)
 	usr, err := user.LookupGroup("gluster")
 	if err != nil {
-		return err
+		if _, ok := err.(user.UnknownGroupError); ok {
+			unknownGrp = true
```

Comment:
You can set `uGID = cuser.Gid` as default and avoid one extra variable(`unknownGrp`). Move `cuser, err := user.Current()` before `LookupGroup()`

Reply:
done